docs/website/contents/for-developers/Genesis.md

```diff
@@ -0,0 +1,718 @@
+  Thus the GDD is not necessarily able to make a decision between them, just as in the previous deadlock.
+
+  A relatively simple fix is suggested by the observation the GDD only compares peers to peers and not to the node's selection.
+  Filling this blindspot suffices to prevent the deadlock.
+  The new logic fits nicely in the ChainSync client, instead of in the GDD itself, since it only involves a single peer at a time.
+
+  If a MsgRollForward arrives that is beyond the forecast range (ie Scg slots beyond the intersection with the node's selection) but the predecessor of the new header (ie the tip of its validated header chain) is not preferable to the selection, then disconnect from the peer.
+  This is effectively a Genesis density comparison between the peer and the syncing node itself (but not necessarily in the GDD window, which is always anchored in the LoE anchor).
+  Even if Sgen were less than Scg, this comparison would still be sufficient and sound, since Sgen = Scg is the most conservative assignment of that Genesis parameter.
+
+  In some sense, this rule is an optimization of the existing ChainSync client logic.
+  It fires as soon as it becomes inevitable that the existing logic will subsequently terminate with ForkTooDeep.
+
+The latter addition also prevents another deadlock.
+If a peer P sends a header beyond the GDD window without having sent any headers within the GDD window, then it must be disconnected from.
+Because Sgen = Scg, such a chain includes a gap at least as large as the forecast range.
+No ChainSync client can progress past such a gap.
+This has always been true, and it has long been identified as a disaster scenario, since it is a severe violation of Chain Growth (PR https://github.com/IntersectMBO/ouroboros-network/pull/2844 is related future work, but remains Closed).
+With CSJ, though, it's now possible that the Dynamo and Objector are stuck on such a gap while some PreObjector would be able to progess if it weren't dormant, just as in the deadlocks discussed above.
+Because this trigger does not require a comparison against another peer, not even Lightweight Checkpointing can prevent it.
+
+The following concrete example of the second deadlock arose in our property tests, inspiring the fix specified above.
+(It could be minimized further, but this one was actually generated.)
+
+```
+    Kcp = 5, Scg = 8, Sgen = 8
+
+    slots:    0  1  2  3  4  5  6  7  8  9 10
+    trunk: G──1─────2──3──4─────5──6──7──8───   Honest    1
+                    ╰──3───────────4──5──6──7   Adversary 2
+                    ╰──3──4────────5─────6──7   Adversary 3
+           ╰─────1──2─────3──4────────5─────6   Adversary 1
+```
+
+Suppose Adversary 1 is the Dynamo first, serves its chain up to the block 5-8, and then the peer disconnects.
+If Adversary 2 or 3 becomes the Dynamo and Adversary 3 or 2 becomes the Objector, then the (HAA-satisfying) honest peer is dormant as a PreObjector.
```

Comment:
Shouldn't we use the same order? Otherwise it's a bit confusing. But maybe there's a reason for inverting the numbers.

```suggestion
If Adversary 2 or 3 becomes the Dynamo and Adversary 2 or 3 becomes the Objector, then the (HAA-satisfying) honest peer is dormant as a PreObjector.
```